sunpy/map/tests/test_plotting.py

```diff
@@ -79,28 +80,48 @@ def test_peek_aia171(aia171_test_map):
 
 @figure_test
 def test_peek_grid_aia171(aia171_test_map):
-    aia171_test_map.peek(draw_grid=True)
+    with warnings.catch_warnings():
+        warnings.filterwarnings("ignore", message=(
+            'The conversion of these 2D helioprojective coordinates to '
+        ), category=SunpyUserWarning)
+        aia171_test_map.peek(draw_grid=True)
```

Comment:
I assume these are rasing this warning because wcsaxes is making multiple calls to the transform stack for each contour line and *some* of them are all NaN?

I am not a fan of the map plotting methods rasing this warning to the user, as there is nothing they can do to stop it. (I.e. the current behaviour is correct and using assume spherical screen would be incorrect). If everything is working as intended maybe we need to silence this warning inside our plotting methods?

Reply:
I can't work out how to silence it.

Reply:
Hmm, this is an interesting pickle.  The reason that you can't squelch the warnings this way is because our logger overrides the normal `warnings` behavior such that any `SunpyUserWarning` gets directed to the logger instead of being handled by the normal filtering machinery.  I don't immediately know what the solution is...

Reply:
I forgot completely/didn't know that we do that with our logger.

Reply:
I imagine that the brute-force solution should work: check if our logger is currently set to capture warnings, and if so, disable it temporarily before entering the `catch_warnings()` context.  You probably should create a convenience context manager for this; I can see this being useful in future areas as well.

That said, I have a suspicion that there may be a problem with some code.  This PR's code is supposed to emit the warning only if an entire line is all NaNs.  But why would any grid line be all NaNs?  By definition every line should all be on disk.  Points at the limb can sometimes be interpreted as off-disk due to numerical precision, but that can't be why an entire line would be all NaNs.  I'm missing something...

Reply:
Okay, I diagnosed this completely wrong: the whole logger business is a red herring.  The good news is that it is possible to filter out the warnings even when the logger is set to capture warnings.  The bad news is that it is highly nontrivial to filter out the warnings within SunPy calls.  That's because the warnings aren't generated by the `draw_grid()` call, which simply instantiates the grid overlay, but rather by the rendering call (e.g., `plt.show()`).  For non-`peek()` plotting, we don't control when rendering happens, so we can't wrap it in a context manager.  The only solution that I can think of is to monkey-patch `WCSAxes`...

Reply:
(Incidentally, the lines of all NaNs are the grid overlay trying to figure out the extent of the map in world coordinates, not any of the grid lines themselves.)